WeatherService/Parsers/IWeatherDataParser.cs

```diff
@@ -0,0 +1,10 @@
+﻿
+using WeatherService.Models;
+
+namespace WeatherService.Parsers
+{
+    public interface IWeatherDataParser
+    {
+        WeatherData Parse(string input);
```

Comment:
let's consider using the Try version. usually, we have methods called TryParse(...) and sometime we also have another version Parse(...). You can check them and see which version is better to be used here. 

```suggestion
        WeatherData TryParse(string input);
```

Reply:
This is a method I created within the class, and I named it Parse. It's a custom one I implemented to deserialize the WeatherData object.